### ISSUE TITLE
Does not mention this depends on q.isPromise

### ISSUE THREAD

**Issue by PyroSA** [bahmutov/time-promise#1]
q.isPromise returns false for a Bluebird promises, so the result is not what is expected.

**Comment by bahmutov** [comment]
Yeah, this is very old code, if you need this feature, I would not mind getting and merging a pull request from you!
